### PR TITLE
Use same color for same namespace.

### PR DESCRIPTION
**Pull request by lchenay**
CF #258

### CODE REVIEW

**Comment by TooTallNate** [comment]
Nice and simple, thanks for the PR! I'm 👍  for this feature. There's been plenty of times where the order of require calls changing ends up changing the colors of the debug instances, since they get instantiated in a different order which has always felt strange.

The only thing I'd be worried about is if the algorithm doesn't "distribute" the colors evenly enough. But I guess we'll just have to feel it out and see.

**Comment by lchenay** [comment]
I agree, it will have less perfect distribution on colors over namespace.
To reduce chance of collision, as it's no more incremental, the only solution is to have more colors available.

File: debug.js [hunk 1]
```diff
@@ -42,13 +42,20 @@ var prevTime;
...
  * @return {Number}
  * @api private
  */
 
-function selectColor() {
-  return exports.colors[prevColor++ % exports.colors.length];
+function selectColor(namespace) {
+  var hash = 0, i, chr, len;
+  
+  for (i in namespace) {
```
Comment (TooTallNate) [inline debug.js]: http://stackoverflow.com/a/7616484/376773

**Comment by Meithar** [comment]
This algorithm doesn't seem to work well. Example:
```js
  console.log({
    namespace,
    hash: Math.abs(hash),
    color: exports.colors[Math.abs(hash) % exports.colors.length],
  });
```
```js
{namespace: "rto-ext:EVENTS", hash: 1808327694, color: "lightseagreen"}
{namespace: "rto-ext:SETTINGS", hash: 1479630660, color: "lightseagreen"}
{namespace: "rto-ext:BG", hash: 33268638, color: "lightseagreen"}
```
Three different namespaces have the same color.

**Comment by lchenay** [comment]
exports.colors.length == 6
So 16% of chance of collision.

@TooTallNate More color ?

Otherwise we need to have an array to store the colors stickiness. Not as complicated as this.

**Comment by lchenay** [comment]
We can try something like that otherwise : https://github.com/lchenay/debug/commit/9f4404f8cd695191743e668ddd0fe4f44ca6cd34

The color will be better distributed as before, with reuse of same color for same namespace, but will use some memory to store the array of mapping.
This can lead to memory leak if a user use debug in prod with random namespace. For exemple those who use a namespace like "myapp:request:${request.expressRequestId}".

**Pull request merged**